test/linalg/rowvector.jl

```diff
@@ -0,0 +1,245 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+@testset "RowVector" begin
+
+@testset "Core" begin
+    v = [1,2,3]
+    z = [1+im,2,3]
+
+    @test RowVector(v) == [1 2 3]
+    @test RowVector{Int}(v) == [1 2 3]
+    @test size(RowVector{Int}(3)) === (1,3)
+    @test size(RowVector{Int}(1,3)) === (1,3)
+    @test size(RowVector{Int}((3,))) === (1,3)
+    @test size(RowVector{Int}((1,3))) === (1,3)
+    @test_throws Exception RowVector{Float64, Vector{Int}}(v)
+
+    @test (v.')::RowVector == [1 2 3]
+    @test (v')::RowVector == [1 2 3]
+    @test (z.')::RowVector == [1+im 2 3]
+    @test (z')::RowVector == [1-im 2 3]
+
+    rv = v.'
+    tz = z.'
+
+    @test (rv.')::Vector == [1, 2, 3]
+    @test (rv')::Vector == [1, 2, 3]
+    @test (tz.')::Vector == [1+im, 2, 3]
+    @test (tz')::Vector == [1-im, 2, 3]
+
+    @test conj(rv) === rv
+    @test conj(tz) == [1-im 2 3]
+
+    @test isa(similar(rv), RowVector)
+    @test isa(similar(rv, Float64), RowVector)
+    @test isa(copy(rv), RowVector)
+
+    @test rv[2] === v[2]
+    @test rv[1,2] === v[2]
+
+    @test (rv2 = copy(rv); rv2[2] = 6; rv2[2] === 6)
+    @test (rv2 = copy(rv); rv2[1,2] = 6; rv2[2] === 6)
+
+    @test length(rv) === 3
+    @test size(rv) === (1,3)
+    @test size(rv,1) === 1
+    @test size(rv,2) === 3
+
+    @test map(-, rv)::RowVector == [-1 -2 -3]
+    @test (-).(rv)::RowVector == [-1 -2 -3]
+    @test (-).(rv,1)::RowVector == [0  1  2]
+
+    y = rand(Complex{Float64},3)
+    @test sum(abs2, imag.(diag(y .+ y'))) < 1e-20
+end
+
+@testset "Diagonal ambiguity methods" begin
+    d = Diagonal([1,2,3])
+    v = [2,3,4]
+    rv = v.'
+
+    @test (rv*d)::RowVector == [2,6,12].'
+    @test_throws Exception d*rv
```

Comment:
Please make these test a specific exception type! This test would pass if your implementation had a typo in it.